lib/stitches/ripple_stitch.py

```diff
@@ -23,34 +25,100 @@ def ripple_stitch(stroke):
     '''
 
     is_linear, helper_lines = _get_helper_lines(stroke)
-    ripple_points = _do_ripple(stroke, helper_lines, is_linear)
+
+    num_lines = len(helper_lines[0])
+    skip_start = _adjust_skip(stroke, num_lines, stroke.skip_start)
+    skip_end = _adjust_skip(stroke, num_lines, stroke.skip_end)
+
+    lines = _get_ripple_lines(stroke, helper_lines, is_linear, skip_start, skip_end)
+    stitches = _get_stitches(stroke, is_linear, lines)
 
     if stroke.reverse:
-        ripple_points.reverse()
+        stitches.reverse()
 
     if stroke.grid_size != 0:
-        ripple_points.extend(_do_grid(stroke, helper_lines))
-
-    stitches = running_stitch(ripple_points, stroke.running_stitch_length, stroke.running_stitch_tolerance)
+        stitches.extend(_do_grid(stroke, helper_lines, skip_start, skip_end))
 
     return _repeat_coords(stitches, stroke.repeats)
 
 
-def _do_ripple(stroke, helper_lines, is_linear):
-    points = []
+def _get_stitches(stroke, is_linear, lines):
+    if is_linear:
+        return _get_staggered_stitches(stroke, lines)
+    else:
+        points = [point for line in lines for point in line]
+        return running_stitch(points, stroke.running_stitch_length, stroke.running_stitch_tolerance)
+
+
+def _get_staggered_stitches(stroke, lines):
+    stitches = []
+    for i, line in enumerate(lines):
+        stitched_line = []
+        connector = []
+        if i != 0 and stroke.join_style == 0:
+            if i % 2 != 0:
```

Comment:
Can we swap the order of the if/else so that the test can be `if i % 2 == 0`?  As it is now the else is a double-negative which made my brain hurt ;)